pkg/controllers/provisioningv2/rke2/machineprovision/controller.go

```diff
@@ -50,13 +51,11 @@ type handler struct {
 func Register(ctx context.Context, clients *wrangler.Context) {
 	h := &handler{
 		ctx: ctx,
-		apply: clients.Apply.
-			WithSetOwnerReference(true, true).
```

Comment:
Why drop this?

Reply:
When this was set, the infrastructure machine object owned the secrets related to provisioning. When the infrastructure object was deleted, the Kubernetes garbage collector would go through and delete all of its dependents. This included the machine state secret.

Then, occasionally, the state secret wouldn't be mounted when the deletion job ran because it wouldn't exists. The job would fail all while keeping the machines in the cloud provider.

By removing this reference, the Kubernetes garbage collector will not clean up the state secret and the job will run successfully. After the job runs, this call will delete the objects that were applied: https://github.com/rancher/rancher/pull/34761/files#diff-11efa64a6ad5ead8500e161819dd3af981e47ea5546f456a25b7a7e18187c47aR240